### PR TITLE
Added multicollinearity checks

### PR DESCRIPTION
**Pull request by TomiSzucs**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Upcoming release (2022)
 =======================
 ## New features
 - Regression analysis
```
Comment (krajcsi) [inline changelog.md:7]:
```suggestion
- Regression analysis (Tamás Szűcs)
```

File: cogstat/cogstat_stat.py [hunk 2]
```diff
@@ -699,6 +699,61 @@ def variable_pair_standard_effect_size(data, meas_lev, sample=True, normality=No
                                                                                  classes="table_cs_pd"))
     return standardized_effect_size_result
 
+def correlation_matrix(data, x):
```
Comment (krajcsi) [inline cogstat/cogstat_stat.py:702]: instead of 'x' a more telling variable name could be used, e.g., variables

File: cogstat/cogstat_stat.py [hunk 3]
```diff
@@ -699,6 +699,61 @@ def variable_pair_standard_effect_size(data, meas_lev, sample=True, normality=No
...
+    -------
+    html text
+    """
+
+    corr_table = data[x].corr()
+    table = _("Pearson correlation matrix of explanatory variables:")
+    table += _format_html_table(corr_table.to_html(bold_rows=False, escape=False, classes="table_cs_pd")) + "\n"
+    return table
+
+def vif_table(data, x):
```
Comment (krajcsi) [inline cogstat/cogstat_stat.py:721]: please use a more explanatory name for x

File: cogstat/cogstat_stat.py [hunk 4]
```diff
@@ -699,6 +699,61 @@ def variable_pair_standard_effect_size(data, meas_lev, sample=True, normality=No
...
+    table = _("Variance inflation factors of explanatory variables and constant")
+    vifs = pd.DataFrame([variance_inflation_factor(X.values, i) \
+                         for i in range(X.shape[1])], index=X.columns, columns=["VIF"])
+    table += _format_html_table(vifs.to_html(bold_rows=False, escape=False, classes="table_cs_pd")) + "\n"
+
+    multicollinearity = False
+    for x_i in x:
+        if vifs.loc[x_i][0] > 10:
+            multicollinearity = True
+            table += "\n" + "<decision>" + _("VIF > 10 in variable ") + x_i + "\n" + _("Possible multicollinearity.")\
```
Comment (krajcsi) [inline cogstat/cogstat_stat.py:747]:
```suggestion
            table += "\n" + "<decision>" + _("VIF > 10 in variable %s ") % x_i + "\n" + _("Possible multicollinearity.")\
```

File: cogstat/cogstat_stat.py [hunk 5]
```diff
@@ -699,6 +699,61 @@ def variable_pair_standard_effect_size(data, meas_lev, sample=True, normality=No
...
+                        + "</decision>"
+            x_other = x.copy()
+            x_other.remove(x_i)
+            slopes = pd.DataFrame(statsmodels.api.OLS(data[x_i], statsmodels.api.add_constant(data[x_other]))\
```
Comment (krajcsi) [inline cogstat/cogstat_stat.py:751]: a short heading or description should be used on the output what this table is

File: cogstat/cogstat_stat.py [hunk 6]
```diff
@@ -699,6 +699,61 @@ def variable_pair_standard_effect_size(data, meas_lev, sample=True, normality=No
...
+
+    Parameters
+    ----------
+    data : pandas dataframe
+    x : list of str
+        list of explanatory variable names
+
+    Returns
+    -------
+    html text, bool
```
[the remark this answers is not in the document]
Comment (TomiSzucs) [reply cogstat/cogstat_stat.py:733]: Docstring updated.

File: cogstat/cogstat_stat.py [hunk 7]
```diff
@@ -699,6 +699,61 @@ def variable_pair_standard_effect_size(data, meas_lev, sample=True, normality=No
...
+    """
+
+    from statsmodels.stats.outliers_influence import variance_inflation_factor
+    X = statsmodels.api.add_constant(data[x])
```
[the remark this answers is not in the document]
Comment (TomiSzucs) [reply cogstat/cogstat_stat.py]: Variable names updated.

File: cogstat/cogstat_stat.py (hunk 2 above)
Comment (TomiSzucs) [reply cogstat/cogstat_stat.py:702]: Variable names updated.

File: cogstat/cogstat_stat.py (hunk 3 above)
Comment (TomiSzucs) [reply cogstat/cogstat_stat.py]: Variable names updated.

File: cogstat/cogstat_stat.py (hunk 5 above)
Comment (TomiSzucs) [reply cogstat/cogstat_stat.py]: Heading added.

**Comment by krajcsi** [comment]
Thank you. Still, some changes are needed, but I'll make them in a follow-up commit.

**Pull request merged**